app/controllers/forgot_passwords_controller.rb

```diff
@@ -0,0 +1,25 @@
+class ForgotPasswordsController < ApplicationController
+  
+  def new; end
+  
+  def create
+    user = User.find_by(email_address: params[:email_address])
+    
+    if user
+      user.generate_token
+      AppMailer.send_forgot_password(user).deliver
+      redirect_to forgot_password_confirmation_path
+    else
+      if params[:email_address].blank?
+        flash[:danger] = "Email cannot be blank. Please try again."
+      else
+        flash[:danger] = "There is no user with that email in the system"
```

Comment:
Be careful with error messages that give out too much information. Many sites (including my own) say something like "If your email address is found in our system, you will receive instructions for resetting your password in a few minutes."